tests/phpunit/integration/Core/Storage/User_TransientsTest.php

```diff
@@ -22,179 +22,166 @@
  */
 class User_TransientsTest extends TestCase {
 
+	use User_Aware_Interface_ContractTests;
+
 	/**
 	 * @var Context
 	 */
 	protected $context;
 
-	use User_Aware_Interface_ContractTests;
+	/**
+	 * @var string
+	 */
+	protected $old_wp_using_ext_object_cache;
 
-	protected function create_user_aware_instance( $user_id ) {
+	public function setUp() {
+		parent::setUp();
 		$this->context = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
-		return new User_Transients( $this->context, $user_id );
 	}
 
-	protected function with_predefined_environment( $callback, $ext_object_cache ) {
-		global $_wp_using_ext_object_cache;
+	public function tearDown() {
+		parent::tearDown();
+		wp_using_ext_object_cache( $this->old_wp_using_ext_object_cache );
+	}
 
-		$old_wp_using_ext_object_cache = $_wp_using_ext_object_cache;
-		$_wp_using_ext_object_cache    = 'with_ext_cache' === $ext_object_cache;
+	protected function create_user_aware_instance() {
+		$user_id = $this->factory()->user->create();
 
-		$user_id         = $this->factory()->user->create();
-		$user_transients = $this->create_user_aware_instance( $user_id );
-		call_user_func( $callback, $user_transients, $user_id );
+		return array(
+			new User_Transients( $this->context, $user_id ),
+			$user_id,
+		);
+	}
 
-		$_wp_using_ext_object_cache = $old_wp_using_ext_object_cache;
+	protected function with_predefined_environment( $ext_object_cache ) {
+		$this->old_wp_using_ext_object_cache = wp_using_ext_object_cache( 'with_ext_cache' === $ext_object_cache );
+		return $this->create_user_aware_instance();
 	}
 
-	protected function using_external_cache( $callback ) {
-		$this->with_predefined_environment( $callback, 'with_ext_cache' );
+	protected function using_external_cache() {
+		return $this->with_predefined_environment( 'with_ext_cache' );
 	}
 
-	protected function using_user_options( $callback ) {
-		$this->with_predefined_environment(
-			function( $user_transients, $user_id ) use ( $callback ) {
-				$user_options = new User_Options( $this->context, $user_id );
-				call_user_func( $callback, $user_transients, $user_options, $user_id );
-			},
-			'without_ext_cache'
+	protected function using_user_options() {
+		list( $user_transients, $user_id ) = $this->with_predefined_environment( 'without_ext_cache' );
```

Comment:
Per my last comment, I think `with_predefined_environment` is no longer necessary since it doesn't need to set the property and `wp_using_ext_object_cache` is quite expressive as it is.

How about this?
```php
protected function using_external_cache() {
	wp_using_ext_object_cache( true );
	return $this->create_user_aware_instance();
}

protected function using_user_options() {
	wp_using_ext_object_cache( false );
	list( $user_transients, $user_id ) = $this->create_user_aware_instance();
	
	return array(
		$user_transients,
		new User_Options( $this->context, $user_id ),
	);
}

Reply:
Updated.